src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -499,6 +507,18 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                 {
                                     if (isRestoreSucceeded)
                                     {
+                                        IEnumerable<NuGetLogCode> nugetLogCodes = restoreSummaries.SelectMany(r => r.Errors.Select(e => e.Code));
+                                        bool shouldDisplayVulnerabilitiesInfoBar = NuGetLogCodeUtility.AreVulnerabilitiesInCodes(nugetLogCodes);
```

Comment:
Given this feedback I decided to remove this public api and go with a method in this class. I got confused with the feedback before but I think this is the best way since there is no requirement of a public api at this time. @zivkan Thanks for the feedback and detailed explanation of the performance!